README.md

```diff
@@ -1,8 +1,118 @@
+ ```  
+specifies that the `breakpoint` config must be distributed to all three available layers: the sass, JavaScript and the templating layer.  
+
+##### Advanced example
+Sometimes configuration values depend on other configuration values, for example measures in a grid system. To be able to maintain these relationships even when the underlying predicate value may be modified by a later-loading config file, the calculation of the final value determined by these relationships must be deferred until all specified configurations are loaded and parsed. This is achieved by specifying these simple mathematical expressions in the format:
+```
+'!expression [some simple mathematical expression]'
+```
+Using this we can specify the baseline grid as:
+```
+config.data.baselinegrid = {space: {}};  
+config.data.baselinegrid.space.extra_small_in_px = 12;  
+config.data.baselinegrid.space.small_in_px = '!expression baselinegrid.space.extra_small_in_px * 2';  
+config.data.baselinegrid.space.smallish_in_px = '!expression baselinegrid.space.small_in_px * 1.5';  
+config.data.baselinegrid.space.medium_in_px = '!expression baselinegrid.space.small_in_px * 2';
+...
+```
+The result is that `config.data.baselinegrid.space.small_in_px` will have the value twice that of whatever the final value of `config.data.baselinegrid.space.extra_small_in_px`is, *even if `config.data.baselinegrid.space.extra_small_in_px` is modified by a later loading config*. This provides a way of reusing the essentials of the baseline grid system, but basing it on a different key value as required.
+
+#### Distributing configuration  
+##### Distributing to SASS  
+Each property of `config.data` specified in `config.layerAllocations.sass` is eventually written as a SASS file to  `/source/css/sass/derived-from-config/_variables--[propertyname].sass`. Each of these files contains the SASS variables describing the config for that property. Looking at the `breakpoint` example again, this config  
+  
+```  
+// specified in a config file  
+config.data.breakpoint = {site: {}};  
+config.data.breakpoint.site.small = 480;  
+config.data.breakpoint.site.medium = 730;  
+  
+config.layerAllocations.sass = ['breakpoint'];  
+```  
+  
+generates this file:  
+```  
+// /source/css/sass/derived-from-config/_variables--breakpoint.sass  
+$breakpoint-site-small: 480;  
+$breakpoint-site-medium: 730;  
+```   
+##### Distributing to JavaScript  
+Each property of `config.data` specified in `config.layerAllocations.jss` is eventually written to `/source/js/derived-from-config/configForJs.json`.  Looking at the `breakpoint` example again, this config:    
+  
+```js  
+// specified in a config file  
+config.data.breakpoint = {site: {}};  
+config.data.breakpoint.site.small = 480;
+config.data.breakpoint.site.medium = 730;  
+config.layerAllocations.js = ['breakpoint'];
+```  
+  
+adds this into `configForJs.json`:  
+```  
+// /source/js/derived-from-config/configForJs.json  
```

Comment:
Pretty wordy, `/source/js/config.json`?

Reply:
Fair point. I like`/source/js/derivedConfig.json`: it indicates that the file shouldn't be directly edited, and we can't insert that information as a comment into the file itself as it's `json`.